ERCS/erc-7660.md

```diff
@@ -0,0 +1,426 @@
+     * @return total The total token will be released in the period.
+     * @return canRelease The current will be released token in the period.
+     * @return released The token has be released in the period.
+     */
+    function getCanReleaseInfo(address account) public view returns (uint256 total, uint256 canRelease, uint256 released);
+
+    /**
+     * @dev Function to claim the user's released token to the account.
+     * @param account The address of the user.
+     */
+    function claimRelease(address account) external;
+     
+    /**
+     * @dev Function to set the accout for exclude vesting.
+     * @param account The address of the sender will exclude vesting or not.
+     * @param flag  The flag whether the sender address will exclude vesting or not.
+     */
+    function setExcludedVest(address account,bool flag) external; 
+    
+    /**
+     * @dev Function to set the uniswap Router address which will receive token without vesting.
+     * @param swapRouter The address of the uniswap router.
+     * @param flag  The flag whether the router address will exclude vesting or not.
+     */
+    function setSwapRouter(address swapRouter, bool flag) external;
+
+    /**
+     * @dev Function to retrieve a user's vest token info in the each circle.
+     * @param account The address of the user.
+     * @param circleIndex The user vest token circle index.
+     * @return total The total token will be released in the period for the user.
+     * @return released  The token has be released in the period for the user.
+     * @return startTime The token will be released at the start time .
+     */
+    function userVestInfo(address account,uint256 circleIndex) external view returns (uint256 total, uint256 released,uint256 startTime);
+}
+```
+
+## Rationale
+
+There are four major improvements in this protocol:
+
+- [ERC-20](./eip-20.md) Tokens and Linear Release Functionality:
+
+  canReleaseAmount = total * (currentTimestamp - startTime) / duration;
+
+
+- Balance is changed by time:
+```solidity
+ function balanceOf(address account) public view virtual override returns (uint256) {
+     (, uint256 canReleaseAmount, ) = getCanReleaseInfo(account,true);
+     return _Owned[account] + canReleaseAmount;
+ }
+
+ function getCanReleaseInfo(address account) public view returns (uint256 total, uint256 canRelease, uint256 released) {
+     for (uint i = 0; i < 7; i++) {
+         VestInfo memory info = userVestInfo[account][i];
+         if (info.startTime == 0) {
+         continue;
+      }
+      released += info.released; 
+      total += info.total;
+      if (block.timestamp <= info.updateTime) {
+         canRelease += info.total; //Just for swap
+      } else if (block.timestamp >= info.startTime + duration) {
+         canRelease += info.total - info.released;
+      } else {
+         uint temp = info.total * (block.timestamp - info.startTime) / duration;
+         canRelease += temp - info.released;
+      }
+    }
+ }
+
+```
```

Comment:
This block is an implementation detail. Only describe the observable behaviours of your contracts, not how to implement them.